[#36] DateSelector 컴포넌트 구현

minyoung1101 opened the pull request:
## 📝 요약
- `DateSelector` 컴포넌트 구현(`DateNav`+`DateTabs`)

## ✨ 구현 내용
- DateNav
  - 현재 선택된 연/월 표시
  - 이전/다음 달 이동 버튼 구현(1일로 이동)
  - 캘린더 버튼 클릭 시 `DatePicker`로 날짜 선택
- DateTabs
  - 선택된 날짜를 중심으로 7일 단위의 날짜 선택 탭 구현
  - framer-motion으로 애니메이션 적용
- DateSelector
  - `DateNav`+`DateTabs`으로 레이아웃 구성
- DatePicker
  - 기존 컴포넌트에서 `selectedDate` 변경 시 월(month)이 변경되지 않는 문제 발견
  - 항상 `selectedDate` 기준으로 렌더링되도록 `defaultMonth` 설정

### 🎯 실제 결과

https://github.com/user-attachments/assets/b3a27d9c-3180-40e1-943a-e79b1ee7a2d3

---

## 💬 리뷰어 참고 사항 및 알게된 점
- ❗chevron icon 12px이 추가되면서 기존 [ large/small ]에서 [ large/medium/small] 으로 옵션이 늘어났습니다
  - chevron 아이콘을 사용하시는 분들은 확인 부탁드립니다!
- 날짜 관련 함수들은 추후 라이브러리 적용 시 수정될 수 있습니다.

## ✅ 체크리스트
- [x] 주요 기능 테스트 완료
- [x] 빌드 및 실행 확인
- [x] 커밋 메시지 컨벤션 및 작업 내용 일치 확인

minyoung1101 committed:
feat [#36] chevron 12px icon 추가 및 icons.ts에 small 사이즈로 등록

minyoung1101 committed:
feat [#36] DateNav 컴포넌트 구현

minyoung1101 committed:
feat [#36] DateTabs 컴포넌트 구현

minyoung1101 committed:
feat [#36] DateSelector 구현(DateNav + DateTabs)

minyoung1101 committed:
feat [#36] DateSelector 스토리북 작성

minyoung1101 committed:
refactor [#36] DatePicker가 selectedDate 기반으로 month를 반영하도록 수정

minyoung1101 committed:
docs [#36] 필요없는 주석 삭제

cskime commented on src/features/tasklist/components/DateNav.tsx:
```diff
@@ -0,0 +1,52 @@
+  selectedDate: Date;
+  onPrev: () => void;
+  onNext: () => void;
+  onOpenCalendar?: () => void;
```
- 개인적으로 `onCalendarOpen` 처럼 동작이 맨 뒤로 나오는게 더 자연스러워 보이네요.
- Event handler 함수를 `handleClickButton`이 아니라 `handleButtonClick`로 쓰는 것과 같습니다.

minyoung1101 replied:
아 그렇네요 
`handle{TARGET}{EVENT}`니까 `onCalendarOpen`이 더 맞는 것 같습니다
감사합니다!

cskime commented on src/features/tasklist/components/DateSelector.tsx:
```diff
@@ -0,0 +1,59 @@
+
+interface DateSelectorProps {
+  selectedDate: Date;
+  onSelectDate: (date: Date) => void;
```
- 여기도 `onOpenCalendar`에 남긴 comment와 같습니다.
- 그런데, 여기서는 `DateSelector`에서 select되는 event handler 이므로 `onSelect`라고만 해도 잘 이해될 것 같네요

minyoung1101 replied:
넵 이 부분도 수정하겠습니다

cskime commented on src/features/tasklist/components/DateSelector.tsx:
```diff
@@ -0,0 +1,59 @@
+            }}
+          />
+        </div>
+      )}
```
- 여유가 되시면 이 부분은 `overlay-kit`을 사용해서 개선해 보면 좋을 것 같습니다.
- `overlay-kit`은 이렇게 `openCalendar` 같은 state 값을 사용한 조건부 렌더링으로 절차적인 코드를 작성하지 않고, React의 선언적 패러다임에 맞게 overlay를 사용할 수 있도록 관리해 줍니다.

minyoung1101 replied:
넵 추후 overlay-kit으로 개선하겠습니당
감사합니다!

cskime reviewed (commented):
시연 영상에서 날짜를 바꿀 때마다 전체적으로 깜빡이는 것 처럼 보이는데, 영상이라 그런걸까요?

minyoung1101 commented:
> 시연 영상에서 날짜를 바꿀 때마다 전체적으로 깜빡이는 것 처럼 보이는데, 영상이라 그런걸까요?

넵 영상 프레임 때문인지 저렇게 나오더라구요ㅠ
로컬에서 실행하거나 스토리북에서 확인했을 때는 끊기는 건 없었습니다!

minyoung1101 committed:
refactor [#36] Date 컴포넌트 props/state 네이밍 정리

- onOpenCalendar → onCalendarOpen
- onSelectedDate → onSelected
- calendarOpen → isCalendarOpen

minyoung1101 committed:
refactor [#36] z-index 999를 커스텀 변수로 사용

ChanMinJeong reviewed (approved):
고생하셨습니다!
